src/values/AbstractObjectValue.js

```diff
@@ -233,9 +245,29 @@ export default class AbstractObjectValue extends AbstractValue {
       invariant(ob2 instanceof ObjectValue);
       let p1 = ob1.$GetPrototypeOf();
       let p2 = ob2.$GetPrototypeOf();
-      let joinedObject = Join.joinValuesAsConditional(this.$Realm, cond, p1, p2);
+      let joinedObject = Join.joinValuesAsConditional(realm, cond, p1, p2);
       invariant(joinedObject instanceof AbstractObjectValue);
       return joinedObject;
+    } else if (this.kind === "explicit conversion to object") {
+      let primitiveValue = this.args[0];
+      switch (primitiveValue.getType()) {
+        case BooleanValue:
+          return realm.intrinsics.BooleanPrototype;
+        case IntegralValue:
+        case NumberValue:
+          return realm.intrinsics.NumberPrototype;
+        case StringValue:
+          return realm.intrinsics.StringPrototype;
+        case SymbolValue:
+          return realm.intrinsics.SymbolPrototype;
```

Comment:
All these cases above should never happen because they should never get an explicit conversion because the wrapper doesn't create an explicit conversion to object. It gets serialized as one of the other object constructors.

https://github.com/facebook/prepack/blob/master/src/methods/to.js#L409-L428

Maybe add an invariant instead to ensure this remains the case?